### DISCUSSION TITLE
[Question]: Deployment on AWS

### DISCUSSION THREAD

**Discussion by rivindubandara** [danny-avila/LibreChat#3745] [Q&A] [answered]
### What is your question?

How can I deploy my version of LibreChat on AWS?

### More Details

I have a customized version of LibreChat that I've just been running locally on my machine through npm. I want to know deploy this onto a web application either through an EC2 instance or a ELB. Can someone explain how this is possible, there seems to be very little documentation around this. Also how would user management work, I understand you can create users profile by registering, but how is that process managed, like resetting passwords or seeing the users who have registered?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I think you should first try setting the app on EC2, it's the same as you would in any other linux environment, assuming you use a distro like Ubuntu: https://www.librechat.ai/docs/remote/docker_linux

**Reply by rivindubandara** [reply]
Hi Danny,

Thanks for this, I seem to have gotten it to work. However none of my customizations seem to have applied. When I run locally it's fine. However in the ec2 instance it's reverted to default. All the code seems to be correct in my repo.

Local host: 
![image](https://github.com/user-attachments/assets/e19d29d6-d63b-43d0-98a7-2c003f413d27)


EC2:
![image](https://github.com/user-attachments/assets/71c229db-1c4e-4bef-aef6-427deab727ec)

In EC2 console:
![image](https://github.com/user-attachments/assets/95367c99-e306-42fe-af99-1de7c4319b3f)

Thanks

**Reply by danny-avila** [reply, marked as answer]
You need to build the image yourself on the EC2 instance. See this comment: https://github.com/danny-avila/LibreChat/discussions/3740#discussioncomment-10423695

Note that 2GB of RAM is usually the minimum for building the docker image.

**Reply by danny-avila** [reply]
2) or, use `npm` and `node` (recommended v10+ and v20+ respectively) on the EC2, also requires 2 GB RAM for compiling frontend code
3) build the docker image somewhere else and pull it from a private container registry (dockerhub, github container registry)

**Reply by rivindubandara** [reply]
@danny-avila I tried the steps in the thread but it's the same result. I have my librechat.yaml config coming in fine, it's just the tailwind css and logo etc which aren't coming through. 

This is what I have in the override file.

![image](https://github.com/user-attachments/assets/18b171cd-6c4d-4832-9958-22d30a6bac64)

**Reply by danny-avila** [reply]
Which compose file are you using?

**Reply by rivindubandara** [reply]
![image](https://github.com/user-attachments/assets/b029e2ac-2c40-4958-9122-5106b20bb712)

This is the docker-compose.yml

**Reply by danny-avila** [reply]
what is the output for this?
```yaml
docker compose down
docker compose build --no-cache
docker compose up -d
```

**Reply by rivindubandara** [reply]
![image](https://github.com/user-attachments/assets/523a2d5c-121e-4272-ac32-941953c90828)

I don't have `docker compose down` on this screenshot but it was done before, just took a long time had to restart console.

**Reply by danny-avila** [reply]
so you managed to build because in the screenshot it doesn't show any build logs?

**Reply by rivindubandara** [reply]
> so you managed to build because in the screenshot it doesn't show any build logs?

Yes `docker compose down` and `docker compose build` both were successful. I meant to say build took a long time when included the local build command.

**Reply by rivindubandara** [reply]
@danny-avila nevermind, changes seem to have taken place. Took quite a while. Thanks for the help

**Reply by AdyaanAkram** [reply]
@rivindubandara please check your email!

**Reply by rivindubandara** [reply]
@danny-avila even though my configs came through, the icon for the model is reverting back for some reason. Works fine in local 
![image](https://github.com/user-attachments/assets/196a0eea-0f45-4480-ab7b-3c00c1ab9dcf)

**Reply by rivindubandara** [reply]
Also wanted to enquire about user management, there doesn't seem to be any docs for it? How do I manage all the users who signup, reset passwords, remove users etc.

**Comment by jls42** [comment]
> ### What is your question?
> How can I deploy my version of LibreChat on AWS?
> 
> ### More Details
> I have a customized version of LibreChat that I've just been running locally on my machine through npm. I want to know deploy this onto a web application either through an EC2 instance or a ELB. Can someone explain how this is possible, there seems to be very little documentation around this. Also how would user management work, I understand you can create users profile by registering, but how is that process managed, like resetting passwords or seeing the users who have registered?
> 
> ### What is the main subject of your question?
> _No response_
> 
> ### Screenshots
> _No response_
> 
> ### Code of Conduct
> * [x]  I agree to follow this project's Code of Conduct

Hello,

Have you tried this one: https://gitlab.com/jls42/poc-librechat-ec2 ? This project provides a basis for deploying LibreChat on an EC2 instance, which could help you get started. Feel free to check the repository's documentation to learn about the specific steps and necessary configurations.

Readme in English : https://gitlab.com/jls42/poc-librechat-ec2/-/blob/main/README-en-gpt-4o.md?ref_type=heads

If you have your own customized version of LibreChat, you might need to adjust the deployment script. For instance, in the user_data.sh script at line 81, replace the default LibreChat repository URL with the URL of your custom version:

Example: Replace the default clone URL with your custom repository URL
git clone https://gitlab.com/your-username/your-custom-librechat.git

Here : https://gitlab.com/jls42/poc-librechat-ec2/-/blob/main/scripts/user_data.sh?ref_type=heads#L81

Good luck with your deployment!

Best regards,
Julien
